src/index.js

```diff
@@ -1,277 +1,234 @@
 
-class Migrate extends Command {
-  static description = 'Upgrade and migrate your Percy SDK to the latest version';
+export const migrate = command('migrate', {
+  description: 'Upgrade and migrate your Percy SDK to the latest version',
+  version: `${pkg.name} ${pkg.version}`,
 
-  static args = [{
-    name: 'sdk_name',
+  args: [{
+    name: 'sdkName',
     description: 'name of the Percy SDK to migrate (detected by default)'
-  }];
-
-  static flags = {
-    // version && help
-    version: flags.version({ char: 'v' }),
-    help: flags.help({ char: 'h' }),
+  }],
 
+  flags: [
     // cli installation
-    'only-cli': flags.boolean({
+    {
+      name: 'only-cli',
       description: 'only run @percy/cli installation'
-    }),
-    'skip-cli': flags.boolean({
+    },
+    {
+      name: 'skip-cli',
       description: 'skip @percy/cli installation'
-    }),
-
-    // logging
-    verbose: flags.boolean({
-      char: 'v',
-      description: 'log everything',
-      exclusive: ['quiet', 'silent']
-    }),
-    quiet: flags.boolean({
-      char: 'q',
-      description: 'log errors only',
-      exclusive: ['verbose', 'silent']
-    }),
-    silent: flags.boolean({
-      description: 'log nothing',
-      exclusive: ['verbose', 'quiet']
-    })
-  };
+    }
+  ],
 
-  static examples = [
+  examples: [
     '$ npx @percy/migrate',
     '$ npx @percy/migrate @percy/puppeteer',
     '$ npx @percy/migrate --only-cli'
-  ];
-
-  // The logger is attached to the default log method to retain internal OCLIF usage
-  log = Object.assign(this.log, logger('migrate'));
-
-  // Initialize flags, args, the loglevel, and attach process handlers for cleanup
-  init() {
-    let { args, flags } = this.parse();
-    this.flags = flags;
-    this.args = args;
-
-    // sets the log level from verbose, quiet, and silent flags
-    logger.loglevel('info', flags);
+  ]
+  // Run migration steps
+}, async function*({ args, flags, log, exit }) {
+  // inspect dependencies
+  let info = await inspectDeps();
+
+  // get the desired sdk migration
+  let sdk = !flags.onlyCli &&
+      await confirmSDK(info, args.sdkName, log);
+
+  // install @percy/cli and migrate config
+  if (!flags.skipCli) {
+    await confirmCLI(info);
+    await confirmConfig(log);
   }
 
-  // Run migration steps
-  async run() {
-    // inspect dependencies
-    let info = inspectDeps();
-
-    // get the desired sdk migration
-    let sdk = !this.flags['only-cli'] &&
-      await this.confirmSDK(info, this.args.sdk_name);
-
-    // install @percy/cli and migrate config
-    if (!this.flags['skip-cli']) {
-      await this.confirmCLI(info);
-      await this.confirmConfig();
+  // perform sdk migration
+  if (sdk?.upgrade) {
+    await confirmUpgrade(sdk);
+    await confirmTransforms(sdk, log);
+    log.info('Migration complete!');
+  } else {
+    if (sdk) {
+      log.warn(
+        `Make sure your SDK is upgraded to the latest version (${sdk.name} ${sdk.version})!`
```

Comment:
We could potentially reuse the command's version here with `migrate.definition.version`

Reply:
Hmm, not sure 🤨 First is `migrate.definition.version` and the second is the actual inspected SDK
<img width="518" alt="image" src="https://user-images.githubusercontent.com/2072894/173136868-778b13e1-9af6-49b5-9546-89ee4e458338.png">

Reply:
Oh duh 🤦 

I saw `${pkg.name} ${pkg.version}` in the command definition and then got the `sdk` variable on this line confused with the `pkg` variable.